fl_training.py

```diff
@@ -82,14 +83,27 @@ def compute_test_score_for_single_node(
 
     # Train model
     print("\n### Training model on one single node: " + str(node.node_id))
-    history = model.fit(
-        node.x_train,
-        node.y_train,
-        batch_size=constants.BATCH_SIZE,
-        epochs=epoch_count,
-        verbose=0,
-        validation_data=(node.x_val, node.y_val),
-    )
+    if is_early_stopping :
+        es = EarlyStopping(monitor='val_loss', mode='min', verbose=0, patience=constants.PATIENCE)
+        history = model.fit(
+            node.x_train,
+            node.y_train,
+            batch_size=constants.BATCH_SIZE,
+            epochs=epoch_count,
+            verbose=0,
+            validation_data=(node.x_val, node.y_val),
+            callbacks=[es,]
+        )   
+    else:
+        history = model.fit(
+            node.x_train,
+            node.y_train,
+            batch_size=constants.BATCH_SIZE,
+            epochs=epoch_count,
+            verbose=0,
+            validation_data=(node.x_val, node.y_val),
+        )
```

Comment:
Just a question because I am wondering: wouldn't it be cleaner to put the `if ... else ...` statement in the content of a variable `callback_list = ...` instead of the whole `model.fit()`?

Reply:
I agree !